### PR TITLE
Fix product-ei/issues/2722

### PR DESCRIPTION
**Pull request by SangeethKarunaratne**
When request is made with json payload the root element had to include the request path.
( request path = request method + request resource )
 eg:   _postemployee , _postemployee_batch_req .
Test cases have been written to ensure that the root element can contain any string and is bypassed to the value contained in the schema.

Resolves  : https://github.com/wso2/product-ei/issues/2722

## Automation tests
 - Unit tests 
   >
 - Integration tests
   > Tested for single dataservice request with json payload
Tested for batch dataservice request with json payload

### CODE REVIEW

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        String postInsertPaymentPayload ="{\n" +
+                "\"employee\":{\n" +
+                "\"employeeNumber\":52,\n" +
+                "\"lastName\":\"Karunaratne\",\n" +
+                "\"firstName\":\"Sangeeth\",\n" +
+                "\"email\":\"sangeeth@wso2.com\",\n" +
+                "\"salary\":18400.00\n" +
+                "}\n" +
+                "}";
+        getHttpResponse(serviceEndPoint+"employee","POST",postInsertPaymentPayload);
```
Comment (GDLMadushanka) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: shall we fix the formatting issue

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        String response = getHttpResponse(serviceEndPoint+"employee/52","GET",null);
+        Assert.assertTrue(response.contains(
```
Comment (GDLMadushanka) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: shall we add an assert message

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                connection.setRequestProperty("Content-Length", String.valueOf(payload.length()));
+                DataOutputStream writer = new DataOutputStream(connection.getOutputStream());
+                writer.write(payload.getBytes());
+                writer.close();
+            }
+            br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            while (null != (line = br.readLine())) {
+                jsonString.append(line);
+            }
+            connection.disconnect();
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: Don't we have to do this in finally block?

Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: fixed

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java (hunk 1 above)
Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: fixed

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
+package org.wso2.ei.dataservice.integration.test.samples;/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: fixed

[the remark this answers is not in the document]
Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: fixed

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class contains test cases to check the json functionality in sending batch request and single request to data service
+ * without wrapping the json payload with _postemployee or _postemployee_batch_req ie: the request path + request method.
+ */
+
```
[the remark this answers is not in the document]
Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: fixed

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/resources/artifacts/DSS/dbs/rdbms/h2/JSONSampleTest.dbs [hunk 6]
```diff
@@ -0,0 +1,101 @@
+<!--
+  ~ Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/resources/artifacts/DSS/dbs/rdbms/h2/JSONSampleTest.dbs]: fixed

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        serviceEndPoint = getServiceUrlHttp(serviceName) + "/";
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        deleteService(serviceName);
+        cleanup();
+    }
+
+    @Test(groups = "wso2.dss", description = "Invoking Insert Request with POST method with JSON payload without _postemployee tag")
```
[the remark this answers is not in the document]
Comment (SangeethKarunaratne) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java]: fixed

**Pull request merged**